lib/services/internalEngine/index.js

```diff
@@ -82,13 +82,27 @@ class InternalEngine {
     if (!this.client) {
       let options;
 
+      // Passed to Elasticsearch's client to make it use
+      // Bluebird instead of ES6 promises
+      const defer = function defer () {
+        let resolve, reject;
+        const
+          promise = new Bluebird((res, rej) => {
+            resolve = res;
+            reject = rej;
+          });
+
+        return {resolve, reject, promise};
+      };
+
       if (this.config.client) {
-        options = Object.assign({}, this.config.client);
+        options = Object.assign({defer}, this.config.client);
       }
       else {
         // keep compatibility layer for the time being
         // @todo: once the install base is clean, remove this
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)